### PR TITLE
Added `service-tags` config option to Consul storage backend

### LINKED ISSUE #1631: Allow additional tags to be added to consul service registration
As the title says enhancment request to add an option to specify additional tags to be added to the consul service registration

### PR DESCRIPTION
**Pull request by vishalnayak**
FIxes #1631

### CODE REVIEW

File: physical/consul.go [hunk 1]
```diff
@@ -108,6 +110,11 @@ func newConsulBackend(conf map[string]string, logger *log.Logger) (Backend, erro
 	}
 	logger.Printf("[DEBUG]: consul: config service set to %s", service)
 
+	// Get the additional tags to attach to the registered service name
+	tags := conf["service-tags"]
+
+	logger.Printf("[DEBUG]: consul: config service-tags set to %s", tags)
```
Comment (jefferai) [inline physical/consul.go]: For clarity can you make this physical/consul?

Comment (vishalnayak) [inline physical/consul.go]: Done. Updated all occurrences.

File: physical/consul.go [hunk 2]
```diff
@@ -548,22 +556,9 @@ func (c *ConsulBackend) reconcileConsul(registeredServiceID string, activeFunc a
 		}
 	}
 
-	tags := serviceTags(active)
-
-	var reregister bool
```
Comment (jefferai) [inline physical/consul.go]: It seems to me like we don't want to give up this logic, although the existing logic could be improved in several ways.

Comment (vishalnayak) [inline physical/consul.go]: Reintroduced the logic and added tags comparison check.

File: helper/strutil/strutil_test.go [hunk 3]
```diff
@@ -2,6 +2,19 @@ package strutil
 
 import "testing"
 
+func TestStrutil_EquivalentSlices(t *testing.T) {
+	slice1 := []string{"test1", "test2", "test3"}
+	slice2 := []string{"test1", "test2", "test3"}
```
Comment (jefferai) [inline helper/strutil/strutil_test.go]: I'd put these out of order between slice1 and slice2, just to test that as well.

**Comment by jefferai** [comment]
That one minor comment on the test, otherwise LGTM!

**Pull request merged**